packages/s/soloud/xmake.lua

```diff
@@ -0,0 +1,62 @@
+-- Examples:
+-- https://github.com/xmake-io/xmake-repo/tree/master/packages/n/newtondynamics
+-- https://github.com/xmake-io/xmake-repo/blob/dev/packages/z/zlib/xmake.lua
+-- https://github.com/tboox/benchbox
+
+package("soloud")
+    set_description("SoLoud is an easy to use, free, portable c/c++ audio engine for games.")
+    
+    add_urls("https://sol.gfxile.net/soloud/soloud_20200207_lite.zip")
+    add_versions("20200207", "a6fa8e5b7d26b03e21947307d45869e33f0d233639258bd5bf4bea73f88e709d")
+    
+    set_homepage("https://sol.gfxile.net/soloud/")
+    set_license("zlib")
+    
+    -- linux needs to link with libpthread and libdl
+    if is_plat("linux") then
+        add_syslinks("pthread", "dl")
+    end
+    
+    on_install(function (package)
+        io.writefile("xmake.lua", [[
+            add_rules("mode.debug", "mode.release")
+            
+            target("soloud")
+                set_kind("$(kind)")
+                set_languages("cxx11")
+                
+                -- for now we'll only support the miniaudio backend
+                add_defines("WITH_MINIAUDIO")
+                
+                add_includedirs("include", {public = true})
+                
+                -- skip `tools` and `backend`
+                add_files("src/**.cpp|tools/**.cpp|backend/**.cpp")
+                add_files("src/**.c|tools/**.c|backend/**.c")
+                -- compile the miniaudio backend
+                -- hide the symbols from the included miniaudio
+                -- to avoid conflicts with xrepo miniaudio.
+                -- we can't use xrepo's miniaudio. it's too new.
```

Comment:
You can select an earlier version of miniaudio if required

Reply:
In xrepo I only see versions 0.11.15, 0.11.16, 0.11.17. This soloud uses 0.10.x.

Reply:
you can add miniaudio  0.10.x  to xmake-repo, then select it.

Reply:
The version of miniaudio needed by soloud (I double-checked. It's v0.9.10 from 2020-01-15) isn't on github. Perhaps this package could be changed to allow the user to choose the backend, since it supports 17. Probably others haven't bit-rotted. I'd like to have this package accepted with its included version of miniaudio, since I've verified that it works on macos, windows, and linux.